dgisim/tests/test_effect.py

```diff
@@ -173,3 +173,57 @@ def testSatiatedEffectRemovedDuringEndRound(self):
             .get_character_statuses()
             .contains(SatiatedStatus)
         )
+    
+    def testEnergyRechargeEffect(self):
+        #set up game
+        game_state = self.ACTION_TEMPLATE.factory().f_player1(
+            lambda p: p.factory().f_characters(
+                lambda cs: cs.factory().active_character_id(1).build()
+            ).build()
+        ).build()
+
+        # initial energy
+        initial_energy = game_state.get_player1().get_characters().get_active_character().get_energy()
```

Comment:
I'm afraid this line cannot pass `mypy` type checks and also IDE type checking. `.get_active_character()` returns `Optional[Character`, which needs to be handled. This is a bit annoying sometimes, but when the game just started, the active character is indeed not chosen thus returns none.
If you are quite confident that it is not None, you can either assert not None before proceeding or make use of `src/helper/quality_of_life/just()`. tbh I don't quite like using just when the expression within is too long, it makes reading the code difficult. I'm still finding alternative ways to get rid of Optional when necessary.